engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/FetchCachedResultsActor.scala

```diff
@@ -0,0 +1,39 @@
+package cromwell.engine.workflow.lifecycle.execution.callcaching
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Props}
+import cromwell.Simpletons._
+import cromwell.core.{DoesNotExpectMessages, JobOutputs}
+import cromwell.core.simpleton.WdlValueBuilder
+import cromwell.database.CromwellDatabase
+import cromwell.database.sql.MetaInfoId
+import cromwell.engine.workflow.lifecycle.execution.callcaching.EngineJobHashingActor.CacheHit
+import cromwell.engine.workflow.lifecycle.execution.callcaching.FetchCachedResultsActor.{CachedOutputLookupFailed, CachedOutputLookupSucceeded}
+import wdl4s.TaskOutput
+
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success}
+
+object FetchCachedResultsActor {
+  def props(cacheHit: CacheHit, taskOutputs: Seq[TaskOutput], replyTo: ActorRef, callCache: CallCache): Props =
+    Props(new FetchCachedResultsActor(cacheHit, taskOutputs, replyTo, callCache))
+
+  sealed trait CachedResultResponse
+  case class CachedOutputLookupFailed(metaInfoId: MetaInfoId, failure: Throwable) extends CachedResultResponse
+  case class CachedOutputLookupSucceeded(jobOutputs: JobOutputs, cacheHit: CacheHit) extends CachedResultResponse
+}
+
+
+class FetchCachedResultsActor(cacheHit: CacheHit, taskOutputs: Seq[TaskOutput], replyTo: ActorRef, callCache: CallCache)
+  extends Actor with ActorLogging with DoesNotExpectMessages with CromwellDatabase {
+  {
```

Comment:
That's just a block that runs at initialization time and prevents any of the contents from being visible outside the class.